### PR TITLE
Verify build fix with prisma field encryption inclusion

### PR DESCRIPTION
**Pull request by BMartinos**
This is a temp fix to see if this will fix the noted buidl issue. prisma field encryption variables seems to be needed during the build process and currently fails without that value being set A short term solution is to populate it is build time, but this needs to be refactored as this presents a securoty issue

## Summary by CodeRabbit

- **New Features**
	- Enhanced security with the addition of environment variables for field encryption.
	- Introduced a new CI/CD job for testing, building, and pushing Docker images based on branch conditions.

- **Chores**
	- Removed outdated test workflow to streamline CI/CD processes.
	- Improved Dockerfile for better build performance and environment configuration.

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -27,6 +28,52 @@ jobs:
...
+          push: true
+          file: docker/production/Dockerfile
+          tags: jembi/smart-health-links-portal:${{ steps.get_branch.outputs.branch_name }}
+
+      - name: Build and push latest
+        if: ${{ steps.get_branch.outputs.branch_name == 'main' }}
+        uses: docker/build-push-action@v5
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
```
Comment (jacob-khoza-symb) [inline .github/workflows/run-tests.yml:73]: Should we not check if it is nit on pull request before we push the image?

Comment (BMartinos) [reply .github/workflows/run-tests.yml:73]: @jacob-khoza-symb Yup thats correct, I still need to make some updates on this action script.
The current changes are just intended for testing purposes to have an image build and sent to dockerhub to verify the build works as intended (without having to merge into master multiple times)

Side note: The entire GHA needs a bit of refactoring and improvement

File: .env.production.example [hunk 2]
```diff
@@ -22,5 +22,5 @@ SHLP_POSTGRES_DB=
 SHLP_POSTGRES_USERNAME=
 SHLP_POSTGRES_PASSWORD=
 POSTGRES_PRISMA_URL=postgresql://${SHLP_POSTGRES_USERNAME}:${SHLP_POSTGRES_PASSWORD}@${SHLP_POSTGRES_HOST}:5432/${SHLP_POSTGRES_DB}
-PRISMA_FIELD_ENCRYPTION_KEY=
-PRISMA_FIELD_ENCRYPTION_HASH_SALT=
+PRISMA_FIELD_ENCRYPTION_KEY=k1.aesgcm256.Wmy6koJaJpjB0H25qBccLfQ84AvpmCjexanqCHJ5Hr0=
+PRISMA_FIELD_ENCRYPTION_HASH_SALT=1a4671c31ff22834a56d4f3741aeac8cc8662a11c73802083ea5a118e5ee600e
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .env.production.example:26]: Agreed, This is not a great approach, but a current limitation being dealt with in the short terms.
To get our production build to succeed, we need these variables available at Docker build time, so a quick solution is to populate a dummy value into the env used in prod (overwritten on docker container startup)

Ideally, this should be moved into a secret and injected at build time, but no time to refactor it at this time. Better solution is to not depend on this env at build time, but rather at run time (future refactor)

File: .github/workflows/tests-build-push.yml [hunk 3]
```diff
@@ -21,23 +38,23 @@ jobs:
...
         uses: docker/setup-buildx-action@v3
 
+      - name: Test Docker build
+        if: ${{ steps.get_branch.outputs.branch_name != 'main' }}
+        uses: docker/build-push-action@v5
+        with:
+          platforms: linux/amd64
+          push: false # dont push image - Used to confirm build success on branches
+          file: docker/production/Dockerfile
+
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .github/workflows/tests-build-push.yml]: fixed in [739005d](https://github.com/jembi/smart-health-links-portal/pull/65/commits/739005da023531ace61e03d7dbd0c38999fc4957)

**Review by jacob-khoza-symb — APPROVED** [review]

**Pull request merged**